Moq.AutoMocker.Generators.Tests/GeneratorsTests.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.Text;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using VerifyCS = Moq.AutoMocker.Generators.Tests.CSharpSourceGeneratorVerifier<Moq.AutoMocker.Generators.UnitTestSourceGenerator>;
+
+namespace Moq.AutoMocker.Generators.Tests;
+
+[TestClass]
+public class GeneratorsTests
+{
+    [TestMethod]
+    public async Task Generation_WithProjectThatDoesNotReferenceAutoMocker_ProducesDiagnosticWarning()
+    {
+        var expectedResult =
+            DiagnosticResult.CompilerWarning(Diagnostics.MustReferenceAutoMock.DiagnosticId);
+        await new VerifyCS.Test
+        {
+            ReferenceAutoMocker = false,
+            ExpectedDiagnostics =
+            {
+                expectedResult
+            }
+        }.RunAsync();
+    }
+
+    [TestMethod]
+    public async Task Generation_WithDecoratedNonPartialClass_ProducesDiagnosticError()
+    {
+        var code = @"
+using Moq.AutoMock;
+
+namespace TestNamespace;
+
+[ConstructorTests(TargetType = typeof(Controller))]
+public class ControllerTests
+{
+    
+}
+
+public class Controller { }
+";
+        var expectedResult =
+            DiagnosticResult.CompilerError(Diagnostics.TestClassesMustBePartial.DiagnosticId)
+                        .WithSpan(6, 1, 10, 2)
+                        .WithArguments("TestNamespace.ControllerTests");
+        await new VerifyCS.Test
+        {
+            TestCode = code,
+            ExpectedDiagnostics =
+            {
+                expectedResult
+            }
+        }.RunAsync();
+    }
+
+    [TestMethod]
+    public async Task Generation_WithNoTargetTypeSpecified_ProducesDiagnosticError()
+    {
+        var code = @"
+using Moq.AutoMock;
+
+namespace TestNamespace;
+
+[ConstructorTests]
+public class ControllerTests
+{
+    
+}
+
+public class Controller { }
+";
+        var expectedResult =
+            DiagnosticResult.CompilerError(Diagnostics.MustSpecifyTargetType.DiagnosticId)
+                        .WithSpan(6, 2, 6, 18)
+                        .WithArguments("TestNamespace.ControllerTests");
+        await new VerifyCS.Test
+        {
+            TestCode = code,
+            ExpectedDiagnostics =
+            {
+                expectedResult
+            }
+        }.RunAsync();
+    }
+
+    [TestMethod]
+    [Description("Issue 142")]
+    public async Task Generation_WithGenericParameter_RemovesInvalidCharactersFromTestsName()
+    {
+        var code = """
```

Comment:
Why raw string literal here and a verbatim string literal in the other tests around it? Just curious